crypto/crypto/src/noise.rs

```diff
@@ -680,8 +685,7 @@ impl NoiseSession {
         nonce.extend_from_slice(&self.read_nonce.to_be_bytes());
         let nonce = GenericArray::from_slice(&nonce);
 
-        let (buffer, authentication_tag) =
-            ciphertext.split_at_mut(ciphertext.len() - AES_GCM_TAGLEN);
+        let (buffer, authentication_tag) = message.split_at_mut(message.len() - AES_GCM_TAGLEN);
```

Comment:
we do use it to calculate `MAX_WRITE_BUFFER_LENGTH` outside of crypto/noise.

I actually like this better inside of crypto/noise itself since we do use AES-GCM directly and this is more explicit : o